src/Coriolis.cpp

```diff
@@ -0,0 +1,147 @@
+#include "RBDyn/Coriolis.h"
+
+namespace rbd
+{
+Eigen::MatrixXd expand(const rbd::Jacobian& jac,
+			const rbd::MultiBody& mb,
+			const Eigen::MatrixXd& jacMat)
+{
+	Eigen::MatrixXd res = Eigen::MatrixXd::Zero(mb.nrDof(), mb.nrDof());
+	expandAdd(jac, mb, jacMat, res);
+	return res;
+}
+
+void expandAdd(const rbd::Jacobian& jac,
+			const rbd::MultiBody& mb,
+			const Eigen::MatrixXd& jacMat,
+			Eigen::MatrixXd& res)
+{
+	int rowJac = 0;
+	int colJac = 0;
+	for(int i : jac.jointsPath())
+	{
+		colJac = 0;
+		for(int j : jac.jointsPath())
+		{
+			res.block(mb.jointPosInDof(i), mb.jointPosInDof(j), mb.joint(i).dof(), mb.joint(j).dof())
+				+= jacMat.block(rowJac, colJac, mb.joint(i).dof(), mb.joint(j).dof());
+			colJac += mb.joint(j).dof();
+		}
+		rowJac += mb.joint(i).dof();
+	}
+}
+
+std::vector<std::array<int, 3>> compactPath(const rbd::Jacobian& jac,
+						const rbd::MultiBody& mb)
+{
+	std::vector<std::array<int, 3>> res;
+
+	int start_block = mb.jointPosInDof(jac.jointsPath()[0]);
+	int len_block = mb.joint(jac.jointsPath()[0]).dof();
+
+	int start;
```

Comment:
Declare inside the loop since it is not meant to be used outside

Reply:
Done